### ISSUE TITLE
Issue with Lint/BooleanSymbol and %i[...]

### ISSUE THREAD

**Issue by marcandre** [rubocop-hq/rubocop#8151]
```sh
$ echo "%i[foo false]" | rubocop -s test -a --only Lint/BooleanSymbol
Inspecting 1 file
W

Offenses:

test:1:8: W: [Corrected] Lint/BooleanSymbol: Symbol with a boolean name - you probably meant to use false.
%i[foo false]
       ^^^^^

1 file inspected, 1 offense detected, 1 offense corrected
====================
%i[foo false]
```

1) Looks like this isn't crashing by chance. There's no spec for this case

2) Autocorrection should not try to autocorrect this case

**Comment by bquorning** [comment]
The fix is pretty easy – just adding e.g. `next unless node.source.include?(':')` right before https://github.com/rubocop-hq/rubocop/blob/b571ba252fba0e13331871b065b28f27c107eeda/lib/rubocop/cop/lint/boolean_symbol.rb#L38

But I cannot seem to write a failing spec – the autocorrector doesn’t loop infinitely in the current version of the code, so maybe the fix should be part of #7868 (where the issue was originally reported)?

**Comment by marcandre** [comment]
If you replace the `delete` by `delete!`, all specs pass. Write a failing spec for that :-)
